src/native/corehost/hostmisc/utils.cpp

```diff
@@ -348,23 +348,25 @@ bool try_stou(const pal::string_t& str, unsigned* num)
     return true;
 }
 
-pal::string_t get_dotnet_root_env_var_name()
+bool get_dotnet_root_from_env(pal::string_t* dotnet_root_env_var_name, pal::string_t* recv)
 {
-    pal::string_t dotnet_root_arch = _X("DOTNET_ROOT_"), path;
-    dotnet_root_arch.append(to_upper(get_arch()));
-    if (pal::getenv(dotnet_root_arch.c_str(), &path))
-        return dotnet_root_arch;
+    *dotnet_root_env_var_name = _X("DOTNET_ROOT_");
+    dotnet_root_env_var_name->append(to_upper(get_arch()));
+    if (get_file_path_from_env(dotnet_root_env_var_name->c_str(), recv))
```

Comment:
If the env. variable is defined but it points to non-existent path this will return false so we will fallback to other env. variables.
We might need something different here.

(I think we should fail if the env. variable is defined but contains invalid path)

Reply:
Agreed.